packages/blade/src/components/Tabs/TabIndicator.native.tsx

```diff
@@ -0,0 +1,41 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import React from 'react';
+import type { TabBarIndicatorProps } from 'react-native-tab-view';
+import { TabBarIndicator as RNTabBarIndicator } from 'react-native-tab-view';
+import { useTabsContext } from './TabsContext';
+import { useTheme } from '~utils';
+
+const TabIndicator = (props: TabBarIndicatorProps<any>): React.ReactElement => {
+  const { theme } = useTheme();
+  const { variant } = useTabsContext();
+  const isFilled = variant === 'filled';
+  return (
+    <RNTabBarIndicator
+      {...props}
+      width="auto"
+      getTabWidth={(index) => {
+        if (!isFilled) return props.getTabWidth(index);
+        if (index === props.navigationState.routes.length - 1) {
+          return props.getTabWidth(index) - theme.spacing[2] * 3;
+        }
+        return props.getTabWidth(index);
+      }}
+      style={{
+        ...(isFilled
+          ? {
+              height: props.layout.height - theme.border.width.thick - theme.spacing[2] * 2,
```

Comment:
I see tab navigation is giving the layout height & width. I hope its well optimised and doesn't cause too many re-renders since its passing layout (that could change) in props 🤔

Reply:
Mostly everything is handled by the library only, and we are already using the [optimization](https://reactnavigation.org/docs/tab-view/#optimize-large-number-of-routes) mentioned in the docs so should be fine.